src/scripts/clipperUI/frontEndGlobals.ts

```diff
@@ -5,25 +5,55 @@ import {Communicator} from "../communicator/communicator";
 import {Logger} from "../logging/logger";
 import {Constants} from "../constants";
 
-export module Clipper {
-	export let injectCommunicator: Communicator;
-	export let extensionCommunicator: Communicator;
-	export let logger: Logger;
-	export let sessionId: SmartValue<string> = new SmartValue<string>();
+import {RemoteStorage} from "../storage/remoteStorage";
+import {StorageAsync} from "../storage/storageAsync";
 
-	export function getUserSessionId(): string {
-		return this.sessionId.get();
+export class Clipper {
+	private static injectCommunicator: Communicator;
+	private static extensionCommunicator: Communicator;
+
+	private static storage: StorageAsync;
+
+	// Public for convenience?
+	public static logger: Logger;
+	public static sessionId: SmartValue<string> = new SmartValue<string>();
+
+	public static getUserSessionId(): string {
+		return Clipper.sessionId.get();
+	}
+
+	public static getInjectCommunicator(): Communicator {
+		return Clipper.injectCommunicator;
+	}
+
+	public static setInjectCommunicator(injectCommunicator: Communicator) {
+		Clipper.injectCommunicator = injectCommunicator;
 	}
 
-	export module Storage {
-		export function getValue(key: string, callback: (value: string) => void): void {
-			Clipper.extensionCommunicator.callRemoteFunction(Constants.FunctionKeys.getStorageValue, { param: key, callback: (value: string) => {
-				callback(value);
-			}});
+	public static getExtensionCommunicator(): Communicator {
+		return Clipper.extensionCommunicator;
+	}
+
+	public static setExtensionCommunicator(extensionCommunicator: Communicator) {
+		Clipper.extensionCommunicator = extensionCommunicator;
+		Clipper.setUpRemoteStorage(extensionCommunicator);
+	}
+
+	public static getValue(key: string, callback: (value: string) => void): void {
```

Comment:
going `Clipper.Storage`  -> `Clipper` in order to abstract away storage makes the purpose of `getValue`/`setValue` feel kind of vague... consider a name change to `getStoredValue`/`storeValue` (or something else more descriptive... not great with naming myself)